src/ol/webgl/TileTexture.js

```diff
@@ -0,0 +1,165 @@
+/**
+ * @module ol/webgl/TileTexture
+ */
+
+import DataTile from '../DataTile.js';
+import EventTarget from '../events/Target.js';
+import EventType from '../events/EventType.js';
+import ImageTile from '../ImageTile.js';
+import TileState from '../TileState.js';
+import WebGLArrayBuffer from './Buffer.js';
+import {ARRAY_BUFFER, STATIC_DRAW} from '../webgl.js';
+import {toSize} from '../size.js';
+
+function bindAndConfigure(gl, texture) {
+  gl.bindTexture(gl.TEXTURE_2D, texture);
+  gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_WRAP_S, gl.CLAMP_TO_EDGE);
+  gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_WRAP_T, gl.CLAMP_TO_EDGE);
+  gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_MIN_FILTER, gl.LINEAR);
+}
+
+/**
+ * @param {WebGLRenderingContext} gl The WebGL context.
+ * @param {WebGLTexture} texture The texture.
+ * @param {HTMLImageElement|HTMLCanvasElement|HTMLVideoElement} image The image.
+ */
+function uploadImageTexture(gl, texture, image) {
+  bindAndConfigure(gl, texture);
+
+  gl.texImage2D(gl.TEXTURE_2D, 0, gl.RGBA, gl.RGBA, gl.UNSIGNED_BYTE, image);
+}
+
+/**
+ * @param {WebGLRenderingContext} gl The WebGL context.
+ * @param {WebGLTexture} texture The texture.
+ * @param {import("../size.js").Size} size The pixel size.
+ * @param {import("../DataTile.js").Data} data The pixel data.
+ */
+function uploadDataTexture(gl, texture, size, data) {
+  bindAndConfigure(gl, texture);
+
+  let format;
+
+  const bytesPerPixel = data.byteLength / (size[0] * size[1]);
+  switch (bytesPerPixel) {
+    case 1: {
+      format = gl.LUMINANCE;
+      break;
+    }
+    case 3: {
```

Comment:
It's possible to use `LUMINANCE_ALPHA` for storing two bytes per texel - inside the GLSL shader, the first byte shall be available by triplicated in the R, G and B components, and the second byte shall be available in the A component. See e.g. https://github.com/openlayers/openlayers/pull/11810/files#diff-3177bea066d4b7181c5fea46bcd65d442c9caf62bcae2ec5042b74eb434948a2R691

Reply:
I know the trick, and used it in an early iteration, but couldn't demonstrate the value to myself.  I'd be curious to hear specifics of the use cases that justify it.